### PR TITLE
Create Unix installer

### LINKED ISSUE #134: Create Unix installer

### PR DESCRIPTION
**Pull request by mkaput**
fix #134

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+  assert_nz "$_installdir" "installdir"
+
+  download "$_resolved_version" "$_arch" "$_installdir" "$_tempdir"
+
+  say "installed scarb to ${_installdir}"
+
+  create_symlink "$_installdir"
+
+  echo
+  if [[ ":$PATH:" != *":${LOCAL_BIN}:"* ]]; then
```
Comment (THenry14) [inline install.sh:93]: Dash does not support`[[ ]]`

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+  local _installdir=$RETVAL
+  assert_nz "$_installdir" "installdir"
+
+  download "$_resolved_version" "$_arch" "$_installdir" "$_tempdir"
+
+  say "installed scarb to ${_installdir}"
+
+  create_symlink "$_installdir"
+
+  echo
```
Comment (THenry14) [inline install.sh:92]: Was this intentional for a newline, or is it here by mistake? :D

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+  if [[ ":$PATH:" != *":${LOCAL_BIN}:"* ]]; then
+    add_local_bin_to_path
+    echo "Then, run 'scarb --version' to verify your installation. Happy coding!"
+  else
+    echo "Scarb has been successfully installed and should be already available in your PATH."
+    echo "Run 'scarb --version' to verify your installation. Happy coding!"
+  fi
+
+  local _retval=$?
```
Comment (THenry14) [inline install.sh:101]: This will always be 0, because it directly follows `echo` command

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,437 @@
...
+
+resolve_version() {
+  local _requested_version=$1
+  local _requested_ref=$2
+
+  local _response
+
+  say "retrieving $_requested_version version from ${SCARB_REPO}..."
+  _response=$(ensure curl -fLs -H 'Accept: application/json' "${SCARB_REPO}/releases/${_requested_ref}")
+  if [ "$_response" == "{\"error\":\"Not Found\"}" ]; then
```
Comment (THenry14) [inline install.sh:356]: It looks like (again, dash) has some problems with it, it looks like it does not support '==' operator

```
$ echo $0
dash
$ response="{\"error\":\"Not Found\"}"
$ if [ "$_response" == "{\"error\":\"Not Found\"}" ]; then echo "noanie"; else echo "nowim"; fi
dash: 9: [: {"error":"Not Found"}: unexpected operator
nowim
$ if [ "$_response" = "{\"error\":\"Not Found\"}" ]; then echo "noanie"; else echo "nowim"; fi
noanie
$
```

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,437 @@
...
+  ensure tar -xz -C "$_installdir" --strip-components=1 -f "$_dl"
+}
+
+create_symlink() {
+  local _installdir=$1
+
+  local _scarb="${_installdir}/bin/scarb"
+  local _symlink="${LOCAL_BIN}/scarb"
+
+  ensure ln -fs "$_scarb" "$_symlink"
```
Comment (THenry14) [inline install.sh:400]: This will fail if $LOCAL_BIN does not exist, I think you should add `mkdir -p` just in case

File: install.sh (hunk 2 above)
Comment (mkaput) [reply install.sh:92]: This is intentional as I wanted to separate logs from final message:

```
➜  scarb git:(mkaput/pkg) ✗ ./install.sh -v0.2.0-alpha.2
scarb-install: retrieving 0.2.0-alpha.2 version from https://github.com/software-mansion/scarb...
scarb-install: downloading scarb-v0.2.0-alpha.2-aarch64-apple-darwin.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 12.5M  100 12.5M    0     0  4107k      0  0:00:03  0:00:03 --:--:-- 5405k
scarb-install: installed scarb to /Users/mk/.local/scarb/0.2.0-alpha.2
scarb-install: created symlink /Users/mk/.local/bin/scarb -> /Users/mk/.local/scarb/0.2.0-alpha.2/bin/scarb

Scarb has been successfully installed and should be already available in your PATH.
Run 'scarb --help' to verify your installation. Happy coding!
```

File: install.sh (hunk 5 above)
Comment (mkaput) [reply install.sh:400]: Done

File: install.sh (hunk 1 above)
Comment (mkaput) [reply install.sh:93]: Fixed

File: install.sh (hunk 4 above)
Comment (mkaput) [reply install.sh:356]: Fixed

File: install.sh (hunk 3 above)
Comment (mkaput) [reply install.sh:101]: Done

**Review by THenry14 — COMMENTED** [review]
lgtm :+1:

**Pull request merged**